ecl/hql/hqlutil.cpp

```diff
@@ -9011,3 +9011,35 @@ IHqlExpression * queryTransformAssignValue(IHqlExpression * transform, IHqlExpre
         return value->queryChild(1);
     return NULL;
 }
+
+//-------------------------------------------------------------------------------------------------
+
+IHqlExpression * convertSetToExpression(bool isAll, size32_t len, const void * ptr, ITypeInfo * setType)
+{
+    HqlExprArray results;
```

Comment:
Because this has been extracted as a general purpose function it requires a little more work - e.g, what about sets of integers etc.  Also, what happens if they pass isAll = true?

I wouldn't implement all variants at this point, but I would throw an error.  So I would add the following:

if (isAll)
   return createValue(no_all, LINK(setType));

and

ITypeInfo * elementType = setType->queryChildType();
switch (elementType->getTypeCode())
{
case type_unicode:
....
case type_string:
....
default:
UNIMPLEMENTED;
}

Then at least anyone else who calls it will quickly discover more code needs implementing.